### PR TITLE
github: Fix attest invalid image name

### PR DESCRIPTION
**Pull request by TimePrinciple**
### Summary of the PR

Strip tag from `IMAGE_NAME` used for attest, which should not contain column and tag. `IMAGE_NAME` is now specified explicitly to `rustvmm/dev`.

### Requirements

Before submitting your PR, please make sure you addressed the following
requirements:

- [x] All commits in this PR have Signed-Off-By trailers (with
  `git commit -s`), and the commit message has max 60 characters for the
  summary and max 75 characters for each description line.
- [x] All added/changed functionality has a corresponding unit/integration
  test.
- [x] All added/changed public-facing functionality has entries in the "Upcoming 
  Release" section of CHANGELOG.md (if no such section exists, please create one).
- [x] Any newly added `unsafe` code is properly documented.

### CODE REVIEW

File: .github/workflows/docker-publish.yml [hunk 1]
```diff
@@ -68,7 +68,7 @@ jobs:
         if: ${{ github.event_name != 'pull_request' }}
         uses: actions/attest-build-provenance@v1
         with:
-          subject-name: index.docker.io/${{ env.VERSION }}
+          subject-name: index.docker.io/rustvmm/dev
```
Comment (stefano-garzarella) [inline .github/workflows/docker-publish.yml:71]: What about add a new command to `docker.sh` and set another env variable in the `Generate next docker tag` step?

Something like this:
```
REGISTRY=$(./docker.sh print-registry)
echo "REGISTRY=${REGISTRY}" >> $GITHUB_ENV
```

Comment (stefano-garzarella) [reply .github/workflows/docker-publish.yml:71]: Or just:
```
echo "REGISTRY=$(./docker.sh print-registry)" >> $GITHUB_ENV
```

Comment (stefano-garzarella) [reply .github/workflows/docker-publish.yml:71]: s/REGISTRY/CONTAINER (or something like that, I was confused, the registry is index.docker.io)

**Review by stefano-garzarella — APPROVED** [review]

File: docker.sh [hunk 2]
```diff
@@ -25,11 +26,19 @@ print_next_version() {
   echo "rustvmm/dev:v$(next_version)"
```
Comment (roypat) [inline docker.sh:26]: could use `$IMAGE_NAME` here:
```suggestion
  echo "$IMAGE_NAME:v$(next_version)"
```

Comment (roypat) [reply docker.sh:26]: Ah, it was like this even before when we had `$DOCKER_TAG`

**Review by roypat — APPROVED** [review]

**Pull request merged**